data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v2/validator/GetAggregateAttestationV2.java

```diff
@@ -0,0 +1,144 @@
+
+public class GetAggregateAttestationV2 extends RestApiEndpoint {
+
+  public static final String ROUTE = "/eth/v2/validator/aggregate_attestation";
+  private final ValidatorDataProvider provider;
+
+  public GetAggregateAttestationV2(
+      final DataProvider dataProvider, final SchemaDefinitionCache schemaDefinitionCache) {
+    this(dataProvider.getValidatorDataProvider(), schemaDefinitionCache);
+  }
+
+  public GetAggregateAttestationV2(
+      final ValidatorDataProvider provider, final SchemaDefinitionCache schemaDefinitionCache) {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("getAggregatedAttestationV2")
+            .summary("Get aggregated attestation")
+            .description(
+                "Aggregates all attestations matching given attestation data root, slot and committee index.\n"
+                    + "A 503 error must be returned if the block identified by the response\n"
+                    + "`beacon_block_root` is optimistic (i.e. the aggregated attestation attests\n"
+                    + "to a block that has not been fully verified by an execution engine).\n"
+                    + "A 404 error must be returned if no attestation is available for the requested\n"
+                    + "`attestation_data_root`.")
+            .tags(TAG_VALIDATOR, TAG_VALIDATOR_REQUIRED)
+            .queryParamRequired(ATTESTATION_DATA_ROOT_PARAMETER)
+            .queryParamRequired(SLOT_PARAMETER)
+            .queryParamRequired(COMMITTEE_INDEX_PARAMETER)
+            .response(
+                HttpStatusCodes.SC_OK, "Request successful", getResponseType(schemaDefinitionCache))
+            .withNotFoundResponse()
+            .build());
+    this.provider = provider;
+  }
+
+  @Override
+  public void handleRequest(final RestApiRequest request) throws JsonProcessingException {
+    final Bytes32 beaconBlockRoot = request.getQueryParameter(ATTESTATION_DATA_ROOT_PARAMETER);
+    final UInt64 slot = request.getQueryParameter(SLOT_PARAMETER);
+    final UInt64 committeeIndex = request.getQueryParameter(COMMITTEE_INDEX_PARAMETER);
+
+    final SafeFuture<Optional<ObjectAndMetaData<Attestation>>> future =
+        provider.createAggregateAndMetaData(slot, beaconBlockRoot, committeeIndex);
+
+    request.respondAsync(
+        future.thenApply(
+            maybeAttestation ->
+                maybeAttestation
+                    .map(
+                        attestationAndMetaData -> {
+                          request.header(
+                              HEADER_CONSENSUS_VERSION,
+                              Version.fromMilestone(attestationAndMetaData.getMilestone()).name());
+                          return AsyncApiResponse.respondOk(attestationAndMetaData);
+                        })
+                    .orElseGet(AsyncApiResponse::respondNotFound)));
+  }
+
+  @SuppressWarnings("unchecked")
+  private static SerializableTypeDefinition<ObjectAndMetaData<Attestation>> getResponseType(
```

Comment:
This seems to be the exact same response as `GetBlockAttestationsV2`. We should probably create a single definition and re-use in both places?

Reply:
I have a feeling we will have this `oneOf(electra, phase0)` logic in other places as well...

Reply:
Ah my bad, I looked into it and for some reason thought it was a list as well!

Reply:
Not sure if `BeaconRestApiTypes` is the right place for it, but nothing that we can't move later if we don't like it I guess.